### PR TITLE
Fork diag/code-action tests to ones without editor dep. 7/N

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/71563

This moves diagnostic config and suppression tests down to the feature-layer tests.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@genlu ptal. 

@dotnet/roslyn-compiler  the only change here is an InternalsVisibleTo to a new test assembly we have.  The compiler already has internalsvisible to a sibling test-assembly.  However, as we are moving these tests to a better location, I needed to update that to now allow access from this new location as well.

**Review by tmat — APPROVED** [review]
:shipit:

File: Roslyn.sln [hunk 1]
```diff
@@ -1633,7 +1633,6 @@ Global
 		src\Compilers\CSharp\csc\CscCommandLine.projitems*{4b45ca0c-03a0-400f-b454-3d4bcb16af38}*SharedItemsImports = 5
 		src\Analyzers\CSharp\Tests\CSharpAnalyzers.UnitTests.projitems*{5018d049-5870-465a-889b-c742ce1e31cb}*SharedItemsImports = 5
 		src\Compilers\CSharp\CSharpAnalyzerDriver\CSharpAnalyzerDriver.projitems*{54e08bf5-f819-404f-a18d-0ab9ea81ea04}*SharedItemsImports = 13
-		src\Analyzers\VisualBasic\Tests\VisualBasicAnalyzers.UnitTests.projitems*{57b7c0aa-e14a-41f6-ad06-fb3937f66fc2}*SharedItemsImports = 5
```
Comment (genlu) [inline Roslyn.sln:1636]: what's this about?

**Review by genlu — APPROVED** [review]

File: Roslyn.sln (hunk 1 above)
Comment (CyrusNajmabadi) [reply Roslyn.sln:1636]: trying to figure it out.  VS keeps adding/removing these items.  it never seems to stabalize.  we pinged project system on another thread about this.

**Pull request merged**